### PR TITLE
Estimation of historical forecast on backtest folds without metrics

### LINKED ISSUE #140: Estimation of historical forecast on backtest folds without metrics
### 🚀 Feature Request

Add posibility to obtain forecast on backtest folds without need to specify metric.

### Proposal

1. Implement `get_historical_forecasts` method for `BasePipeline`
1.1. Method signature should be similar to `BasePipeline.backtest` signature
1.2. Call  `BasePipeline.backtest` inside method with `_DummyMetric`
1.3. Use `tslogger.disable` context manager to turn off logging while calling `backtest`
1.4. Return dataframe with forecasts on folds
2. Update `NaiveVariance._compute_resids_matrices` to use new method
3. Update `BasePipeline._forecast_prediction_interval` to use new method

Proposed `get_historical_forecasts` signature:
```python
def get_historical_forecasts(
        self,
        ts: TSDataset,
        n_folds: Union[int, List[FoldMask]] = 5,
        mode: Optional[str] = None,
        n_jobs: int = 1,
        refit: Union[bool, int] = True,
        stride: Optional[int] = None,
        joblib_params: Optional[Dict[str, Any]] = None,
        forecast_params: Optional[Dict[str, Any]] = None,
    ) -> pd.DataFrame:
```

### Test cases

Add test cases similar to `backtest` fold forecast tests.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by brsnw250**
## Before submitting (must do checklist)

- [x] Did you read the [contribution guide](https://github.com/etna-team/etna/blob/master/CONTRIBUTING.md)?
- [x] Did you update the docs? We use Numpy format for all the methods and classes.
- [x] Did you write any new necessary tests?
- [x] Did you update the [CHANGELOG](https://github.com/etna-team/etna/blob/master/CHANGELOG.md)?

## Closing issues

closes #140

### CODE REVIEW

File: tests/test_pipeline/test_pipeline.py [hunk 1]
```diff
@@ -757,6 +823,15 @@ def test_backtest_forecasts_sanity(step_ts: TSDataset):
     assert np.all(forecast_df == expected_forecast_df)
 
 
+def test_get_historical_forecasts_sanity(step_ts: TSDataset):
```
Comment (d-a-bunin) [inline tests/test_pipeline/test_pipeline.py:826]: Could it be `test_sanity_get_historical_forecasts`?

File: tests/test_pipeline/test_pipeline.py [hunk 2]
```diff
@@ -1128,6 +1216,13 @@ def test_forecast_backtest_correct_ordering(step_ts: TSDataset):
     assert np.all(forecast_df.values == expected_forecast_df.values)
 
 
+def test_get_historical_forecasts_correct_ordering(step_ts: TSDataset):
```
Comment (d-a-bunin) [inline tests/test_pipeline/test_pipeline.py:1219]: Is this test really about ordering? I can see comparing raw values.

**Review by d-a-bunin — CHANGES_REQUESTED** [review]
Small questions.

File: tests/test_pipeline/test_pipeline.py (hunk 2 above)
Comment (brsnw250) [reply tests/test_pipeline/test_pipeline.py:1219]: Same naming as in the test above for `backtest` to show it is the similar test.

File: tests/test_pipeline/test_pipeline.py (hunk 1 above)
Comment (brsnw250) [reply tests/test_pipeline/test_pipeline.py:826]: We could, but it just follows existing naming (e.g. `test_backtest_forecasts_sanity` above)

**Review by d-a-bunin — APPROVED** [review]

**Pull request merged**